sdk/servicebus/service-bus/src/sender.ts

```diff
@@ -157,14 +164,15 @@ export class ServiceBusSenderImpl implements ServiceBusSender {
    * @throws Error if the underlying connection is closed.
    */
   constructor(
-    clientId: string,
     private _context: ConnectionContext,
     private _entityPath: string,
-    retryOptions: RetryOptions = {}
+    retryOptions: RetryOptions = {},
+    identifier?: string
   ) {
     throwErrorIfConnectionClosed(_context);
     this.entityPath = _entityPath;
-    this._sender = MessageSender.create(clientId, this._context, _entityPath, retryOptions);
+    this.identifier = identifier ? identifier : `${this.entityPath}-${generate_uuid()}`;
```

Comment:
should this logic be factored out into its own function and be used in all places that generate that identifier to ensure uniformity?

Reply:
Refactor into `ensureValidIdentifier()` helper function.